extensions/amp-smartlinks/0.1/test/test-amp-smartlinks.js

```diff
@@ -0,0 +1,247 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as DocumentReady from '../../../../src/document-ready';
+import * as LinkmateOptions from '../linkmate-options';
+import {AmpSmartlinks} from '../amp-smartlinks';
+import {LinkRewriterManager} from
+  '../../../amp-skimlinks/0.1/link-rewriter/link-rewriter-manager';
+import {Services} from '../../../../src/services';
+
+const helpersFactory = env => {
+  return {
+    createAmpSmartlinks(extensionAttrs) {
+      const ampTag = document.createElement('amp-smartlinks');
+
+      for (const attr in extensionAttrs) {
+        ampTag.setAttribute(attr, extensionAttrs[attr]);
+      }
+      ampTag.getAmpDoc = () => env.ampdoc;
+
+      return new AmpSmartlinks(ampTag);
+    },
+  };
+};
+
+describes.fakeWin('amp-smartlinks',
+    {amp: {extensions: ['amp-smartlinks']}},
+    env => {
+      let ampSmartlinks, helpers, xhr;
+
+      beforeEach(() => {
+        xhr = Services.xhrFor(env.win);
+        helpers = helpersFactory(env);
+        env.sandbox
+            .stub(DocumentReady, 'whenDocumentReady')
+            .returns(Promise.reject());
+      });
+
+      afterEach(() => {
+        env.sandbox.restore();
+      });
+
+      describe('getConfigOptions', () => {
+        it('Should parse options', () => {
+          env.sandbox.spy(LinkmateOptions, 'getConfigOptions');
+
+          const smartlinkOptions = {
+            'nrtv-account-name': 'thisisnotapublisher',
+            'linkmate': true,
```

Comment:
@PhilWinchester Since we change the validation rule, the test files will need to be updated as well.

Reply:
@zhouyx I wasn't sure the best way to mock an empty attribute tag in this dict so I went with empty strings. Let me know if that's an issue for this test.

Reply:
empty string should be fine.